### PR TITLE
feat: state proof

### PR DESCRIPTION
**Pull request by vuvoth**
- State proof native rust impl 
- Test base on real data

Notes: 
- Visibility here is public for testing. We will update it better after merge.

### CODE REVIEW

**Comment by vuvoth** [comment]
Let review after #3

File: sui-execution/latest/sui-move-natives/src/light_client/tendermint_lc.rs [hunk 1]
```diff
@@ -46,9 +56,49 @@ pub fn tendermint_state_proof(
     ty_args: Vec<Type>,
     mut args: VecDeque<Value>,
 ) -> PartialVMResult<NativeResult> {
-    todo!()
+    let value = pop_arg!(args, Vector).to_vec_u8()?;
```
Comment (robert-zaremba) [inline sui-execution/latest/sui-move-natives/src/light_client/tendermint_lc.rs:55]: can we use "destructing" pattern? See:
* https://doc.rust-lang.org/rust-by-example/flow_control/match/destructuring/destructure_slice.html
* https://stackoverflow.com/a/72552078

File: sui-execution/latest/sui-move-natives/src/light_client/tendermint_lc.rs [hunk 2]
```diff
@@ -46,9 +56,49 @@ pub fn tendermint_state_proof(
...
+    let path = pop_arg!(args, Vector).to_vec_u8()?;
+    let prefix = pop_arg!(args, Vector).to_vec_u8()?;
+    let root = pop_arg!(args, Vector).to_vec_u8()?;
+    let proof = pop_arg!(args, Vector).to_vec_u8()?;
+
+    let Ok(prefix) = CommitmentPrefix::try_from(prefix) else {
+        return Ok(NativeResult::err(context.gas_used(), 1));
```
Comment (robert-zaremba) [inline sui-execution/latest/sui-move-natives/src/light_client/tendermint_lc.rs]: please don't use magic number (1) -- define constants. Make sure we have all error codes unique in the whole package.

Comment (robert-zaremba) [reply sui-execution/latest/sui-move-natives/src/light_client/tendermint_lc.rs]: so probably we should have `light_client/errors.rs` module.

**Review by robert-zaremba — APPROVED** [review]
pre-approving

File: sui-execution/latest/sui-move-natives/src/light_client/tendermint_lc.rs (hunk 1 above)
Comment (vuvoth) [reply sui-execution/latest/sui-move-natives/src/light_client/tendermint_lc.rs:55]: I don't think so and we need make our code consistency with sui/dwallet also.

File: sui-execution/latest/sui-move-natives/src/light_client/tendermint_lc.rs (hunk 2 above)
Comment (vuvoth) [reply sui-execution/latest/sui-move-natives/src/light_client/tendermint_lc.rs]: Yes! I add todo for this. Will do it after merge,.

**Pull request merged**